src/wallet/wallet.cpp

```diff
@@ -3299,6 +3299,9 @@ bool CWallet::InitLoadWallet()
             key.MakeNewKey(true);
             if (!walletInstance->SetHDMasterKey(key))
                 throw std::runtime_error("CWallet::GenerateNewKey(): Storing master key failed");
+
+            // ensure this wallet.dat can only be opened by clients supporting HD
+            walletInstance->SetMinVersion(FEATURE_HD);
```

Comment:
I think this call should go above SetHDMasterKey...cant have it crash in between and leave the wallet with an HD seed but no minversion set (not that it really matters, but, you know....)